examples/ASDKgram/Sample/CommentFeedModel.m

```diff
@@ -0,0 +1,196 @@
+{
+  return [_comments objectAtIndex:index];
+}
+
+- (NSUInteger)numberOfCommentsForPhoto
+{
+  return _totalItems;
+}
+
+- (BOOL)numberOfCommentsForPhotoExceedsInteger:(NSUInteger)number
+{
+  return (_totalItems > number);
+}
+
+- (NSAttributedString *)viewAllCommentsAttributedString
+{
+  NSString *string               = [NSString stringWithFormat:@"View all %@ comments", [NSNumber numberWithUnsignedInteger:_totalItems]];
+  NSAttributedString *attrString = [NSAttributedString attributedStringWithString:string fontSize:14 color:[UIColor lightGrayColor] firstWordColor:nil];
+  return attrString;
+}
+
+- (void)requestPageWithCompletionBlock:(void (^)(NSArray *))block
+{
+  // only one fetch at a time
+  if (_fetchPageInProgress) {
+      return;
+  } else {
+    _fetchPageInProgress = YES;
+    [self fetchPageWithCompletionBlock:block];
+  }
+}
+
+- (void)refreshFeedWithCompletionBlock:(void (^)(NSArray *))block
+{
+  // only one fetch at a time
+  if (_refreshFeedInProgress) {
+    return;
+  } else {
+    _refreshFeedInProgress = YES;
+    _currentPage = 0;
+    
+    // FIXME: blow away any other requests in progress
+    
+    [self fetchPageWithCompletionBlock:^(NSArray *newPhotos) {
+      if (block) {
+        block(newPhotos);
+      }
+      _refreshFeedInProgress = NO;
+    } replaceData:YES];
+  }
+}
+
+#pragma mark - Helper Methods
+- (void)fetchPageWithCompletionBlock:(void (^)(NSArray *))block
+{
+  [self fetchPageWithCompletionBlock:block replaceData:NO];
+}
+
+- (void)fetchPageWithCompletionBlock:(void (^)(NSArray *))block replaceData:(BOOL)replaceData
+{
+  // early return if reached end of pages
+  if (_totalPages) {
+    if (_currentPage == _totalPages) {
+      return;
+    }
+  }
+  
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    
+    NSMutableArray *newComments = [NSMutableArray array];
+    
+    @synchronized(self) {
+      
+      NSUInteger nextPage = _currentPage + 1;
+      
+      NSString *urlAdditions = [NSString stringWithFormat:@"page=%lu", (unsigned long)nextPage];
+      NSURL *url = [NSURL URLWithString:[_urlString stringByAppendingString:urlAdditions]];
+      
+      NSData *data = [NSData dataWithContentsOfURL:url];
+      
+      if (data) {
+        
+        NSDictionary *response = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+        
+        if ([response isKindOfClass:[NSDictionary class]]) {
+          
+          _currentPage = [[response valueForKeyPath:@"current_page"] integerValue];
+          _totalPages  = [[response valueForKeyPath:@"total_pages"] integerValue];
+          _totalItems  = [[response valueForKeyPath:@"total_items"] integerValue];
+          
+          NSArray *comments = [response valueForKeyPath:@"comments"];
+          
+          if ([comments isKindOfClass:[NSArray class]]) {
+            
+            NSUInteger numComments = [comments count];
+            if (numComments > 3) {
```

Comment:
If you email @appleguy at AsyncDisplayKit@gmail.com he can add you to the ASDK slack channel. I’m sure he'll be able to give you a much more complete comparison than my app currently does. I’m also helping upgrade the ASDK documentation right now, so if you have any questions as you put together your presentation, feel free to ping me on slack (@hannah).

- As far as my app goes, since it hasn't been fully vetted by @appleguy yet, I'm not sure it is a completely fair comparison of ASDK vs UIKit (but I hope to soon make it so). 
- I think that I need to make it so that the dataModel clears itself when switching between tabs. As it stands, it isn’t a fair comparison, because the app opens on the ASDK tab and by the time you switch to the UIKit tab, the dataModel has already downloaded a bunch of data.  
- I plan on implementing a collectionView in the next few days which will much more dramatically compare ASDK vs UIKit. 
- I also own a 6s+ and yes, it is difficult to perceive the difference on a high speed internet connection with a top of the line apple device. However, if you have a slower device (the 5th Gen iPod is the ideal test device according to @appleguy), like what much of the general population owns, you can much more easily see the difference. 
- Check out https://github.com/kconner/KMCGeigerCounter. With a one line change in the AppDelegate, you can see / hear the frame drops on each home feed. 

Another interesting point that you might want to research for your presentation is this: ASDK was built for performance, however, in doing research for the documentation for the 2.0 launch, the features that come with ASDK are compelling enough set to adopt the framework on their own! There are far too many to mention here, but the ASRangeController (intelligent preloading), automatic batch fetching, concurrent layout, high performance corner rounding, HTTP live video streaming, ridiculous tappable text support (across line breaks), extensible and reusable layout,… the list goes on and on. :)